core/sdk-core/src/test/java/software/amazon/awssdk/core/sync/RequestBodyTest.java

```diff
@@ -104,4 +103,19 @@ public void emptyBytesConstructorHasCorrectContentType() {
         RequestBody requestBody = RequestBody.empty();
         assertThat(requestBody.contentType()).isEqualTo(Mimetype.MIMETYPE_OCTET_STREAM);
     }
+
+    @Test
+    public void remainingByteBufferConstructorOnlyRemainingBytesCopied() throws IOException {
+        ByteBuffer bb = ByteBuffer.allocate(4);
+        bb.putInt(42);
+        bb.flip();
+
+        bb.get();
+
+        RequestBody requestBody = RequestBody.fromRemainingByteBuffer(bb);
+        assertThat(requestBody.contentLength()).isEqualTo(bb.remaining());
```

Comment:
Capture bb.remaining() before calling fromRemainingByteBuffer just to ensure that it's not being mutated.

Reply:
Or have a separate test to assert such...

Reply:
Will do